pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/UseUtilityClassRule.java

```diff
@@ -84,7 +84,7 @@ public Object visit(ASTClassOrInterfaceBody decl, Object data) {
     private boolean isOkUsingLombok(ASTClassOrInterfaceDeclaration parent) {
         // check if there's a lombok no arg private constructor, if so skip the rest of the rules
         if (hasClassLombokAnnotation()) {
-            ASTAnnotation annotation = getLombokAnnotation(parent, "NoArgsConstructor");
+            ASTAnnotation annotation = parent.getAnnotation("lombok.NoArgsConstructor");
 
             if (annotation != null) {
```

Comment:
this double check for lombok annotation + specific lombok annotation seems pointless, just keep the second (specific) check

Reply:
> Maybe directly extending AbstractIgnoredAnnotRule and settig the default property to lombok.NoArgsConstructor?

I agree this one since this class extend `AbstractLombokAwareRule` was made in the [commit](https://github.com/pmd/pmd/commit/9938a4491c510dcb6102388b4f70bcbdde0d62e7#diff-4cadad203122382dac96b07b35245430R23) after last release. So we don't have to worry about breaking API change.

Reply:
@jsotuyod 
The ignored annotation in this rule is special since the rule also check` accessl.level` of the `NoArgsConstructor` annotation. And the rule should't allow user to setup the ignoredAnnotations property. So, this rule should't extend `AbstractLombokAwareRule` or `AbstractIgnoredAnnotRule`.